### PR TITLE
Sync folder item iterator

### PR DESCRIPTION
**Pull request by ScottDugas**
This calls sync folder items multiple times, until it gets all ids for the given folder.

### CODE REVIEW

File: src/main/java/com/ripariandata/timberwolf/exchange/SyncFolderItemIterator.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            }
+            if (retrievedLastItem)
+            {
+                return null;
+            }
+            SyncFolderItemsHelper.SyncFolderItemsResult result =
+                    SyncFolderItemsHelper.syncFolderItems(service, config, folder);
+            syncState = result.getSyncState();
+            LOG.debug("Got {} email ids, which was {}the last them.", result.getIds().size(),
+                      result.includesLastItem() ? "" : "not ");
```
Comment (skermes) [inline src/main/java/com/ripariandata/timberwolf/exchange/SyncFolderItemIterator.java:72]: There's some real interesting grammar going on here.  Maybe you're looking for something like `"Got {} email ids, which were {}the last of them."`

File: src/main/java/com/ripariandata/timberwolf/exchange/SyncFolderItemIterator.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            retrievedLastItem = result.includesLastItem();
+            if (result.getIds().size() > 0)
```
Comment (skermes) [inline src/main/java/com/ripariandata/timberwolf/exchange/SyncFolderItemIterator.java:74]: What's the relationship between this predicate and `includesLastItem` ?

File: src/main/java/com/ripariandata/timberwolf/exchange/SyncFolderItemsHelper.java [hunk 3]
```diff
@@ -111,15 +113,11 @@ private static SyncFolderItemsResult syncFolderItems(final ExchangeService excha
             }
             if (message.isSetSyncState())
             {
-                folder.setSyncStateToken(message.getSyncState());
+                result.setSyncState(message.getSyncState());
             }
             if (message.isSetIncludesLastItemInRange())
```
Comment (skermes) [inline src/main/java/com/ripariandata/timberwolf/exchange/SyncFolderItemsHelper.java:118]: IncludesLastItemInRange can never be false?

File: src/test/java/com/ripariandata/timberwolf/exchange/ExchangeMailStoreTest.java [hunk 4]
```diff
@@ -27,32 +27,24 @@
 import com.microsoft.schemas.exchange.services.x2006.types.FolderIdType;
 import com.microsoft.schemas.exchange.services.x2006.types.FolderType;
 import com.microsoft.schemas.exchange.services.x2006.types.MessageType;
-
```
Comment (skermes) [inline src/test/java/com/ripariandata/timberwolf/exchange/ExchangeMailStoreTest.java:30]: What's the correct thing to do w.r.t spaces between import groups?  I thought we were supposed to be putting a blank line, here but this is at least the second time I've read a diff that removed a whole bunch of them.

**Comment by ScottDugas** [comment]
I also made that change to the wording, but committed it after the merge

**Pull request merged**